apps/meteor/tests/unit/client/views/admin/rooms/RoomsTable.spec.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-restricted-properties */
+import { RoomType } from '@rocket.chat/apps-engine/definition/rooms';
+import * as Hooks from '@rocket.chat/fuselage-hooks';
+import * as Contexts from '@rocket.chat/ui-contexts';
+import * as ReactQuery from '@tanstack/react-query';
+import '@testing-library/jest-dom';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { render, screen } from '@testing-library/react';
+import { expect } from 'chai';
+import React from 'react';
+import sinon from 'sinon';
+
+import RoomsTable from '../../../../../../client/views/admin/rooms/RoomsTable';
+
+jest.mock('../../../../../../client/views/admin/rooms/RoomRow', () => {
+	return ({ someProp }: any) => (
+		<tr data-id='RoomRow'>
+			<td>MockedRoomRow {someProp}</td>
+		</tr>
+	);
+});
+
+const createQueryClientWrapper = () => {
+	const queryClient = new QueryClient();
+	const provider = ({ children }: any) => {
+		return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
+	};
+	provider.displayName = 'provider';
+
+	return provider;
+};
+
+describe('RoomsTable', () => {
+	let useQueryStub: any;
+
+	beforeEach(() => {
+		sinon.stub(Contexts, 'useEndpoint').returns(
+			sinon.fake(() =>
+				Promise.resolve({
+					data: { rooms: [], total: 0 },
+					isSuccess: true,
+					isLoading: false,
+					isError: false,
+					refetch: sinon.fake(),
+				} as never),
+			),
+		);
+		useQueryStub = sinon.stub(ReactQuery, 'useQuery').returns({} as any);
+		sinon.stub(Hooks, 'useMediaQuery').returns(true);
+		sinon.stub(Hooks, 'useLocalStorage').returns([[], sinon.fake()]);
+		sinon.stub(Hooks, 'useDebouncedValue').callsFake((value) => value);
+		sinon.stub(Contexts, 'useTranslation').returns({} as any);
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	it('renders loading state initially', async () => {
+		useQueryStub.returns({
+			isLoading: true,
+			data: undefined,
+			error: undefined,
+		});
+
+		render(<RoomsTable reload={{ current: sinon.fake() }} />, { wrapper: createQueryClientWrapper() });
+
+		expect(screen.queryByTestId('RoomGenericTableLoadingTable')).not.to.be.undefined;
+
+		sinon.restore();
+	});
+
+	it('renders no results when data is empty', async () => {
+		useQueryStub.returns({
+			isLoading: false,
+			data: {
+				rooms: [],
+			},
+			error: undefined,
+			isSuccess: true,
+		});
+		render(<RoomsTable reload={{ current: sinon.fake() }} />, { wrapper: createQueryClientWrapper() });
```

Comment:
Since you're passing fakes to the calls, maybe you can assert those are being called on the proper places too? 👀